src/test/java/seedu/address/logic/commands/DeleteProcCommandTest.java

```diff
@@ -1,29 +1,89 @@
 package seedu.address.logic.commands;
 
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.Assert.assertThrows;
 import static seedu.address.testutil.TypicalClients.getTypicalAddressBook;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_CLIENT;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_PROCEDURE;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_CLIENT;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND_PROCEDURE;
 
+import java.util.ArrayList;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.AddressBook;
 import seedu.address.model.Model;
 import seedu.address.model.ModelManager;
 import seedu.address.model.UserPrefs;
-
+import seedu.address.model.client.Client;
+import seedu.address.model.procedure.Procedure;
+import seedu.address.testutil.ClientBuilder;
+import seedu.address.testutil.ProcedureBuilder;
 
 /**
  * Contains integration tests (interaction with the Model) and unit tests for EditCommand.
  */
 public class DeleteProcCommandTest {
 
     private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
-    //Will do up more in due time!
-    //@Test
-    //public void execute_procedureWasDeletedFromClient_success() {
-    //Client editedClient = new ClientBuilder().build();
-    //DeleteProcCommand deleteCommand = new DeleteProcCommand(INDEX_FIRST_CLIENT, INDEX_FIRST_PROCEDURE);
-
-    //String expectedMessage = String.format(DeleteProcCommand.MESSAGE_EDIT_CLIENT_SUCCESS,
-    // editedClient.getProcedures());
-    //Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
-    //expectedModel.setClient(model.getFilteredClientList().get(0), editedClient);
-
-    //assertCommandSuccess(deleteCommand, model, expectedMessage, expectedModel);
-    //}
+
+    @Test
+    public void execute_procedureDeletedFromExistingClient_success() {
+        Procedure validProcedure = new ProcedureBuilder().build();
+        ArrayList<Procedure> validProcedureList = new ArrayList<>();
+        validProcedureList.add(validProcedure);
+
+        Client clientToEdit = model.getFilteredClientList().get(INDEX_FIRST_CLIENT.getZeroBased());
+        Client validClient = new ClientBuilder(clientToEdit).withProcedures(validProcedureList).build();
+        model.setClient(clientToEdit, validClient);
```

Comment:
I think it should be the `expectedModel` that should be directly accessed rather than `model`.

From what I know, `assertCommandSuccess` runs the command based on the model and compares its execution with the expectedModel.

Reply:
Hihi JB, yes you are right! :) Currently, my implementation for the model is to add the Procedure to the model, so that it can be deleted later during the command and compared to that of the expectedModel, where the Client will no longer have any Procedures then. Hence, I did not edit the expectedModel, because this is a deleteCommand, and the client should not have any Procedures after.

Reply:
Hi Ben, thank you for such detailed comment! Yes I was reading it thinking it was addProcCommand... must have mixed it up while reviewing your code. Your testing is absolutely right!!!